### PR TITLE
[hotfix][state] Remove hardcoded backend class name check

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

A trivial refactoring to remove hardcoded class check.

## Verifying this change

Covered by existing `StateConfigUtilTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by Myasuka** [comment]
LGTM, +1 to merge.

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java [hunk 1]
```diff
@@ -349,7 +349,12 @@ public int numKeyValueStatesByName() {
     }
 
     // TODO remove this once heap-based timers are working with RocksDB incremental snapshots!
-    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointOptions) {
+    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointType) {
+        return false;
+    }
+
+    // TODO remove this once heap-based timers are working with RocksDB incremental snapshots!
```
Comment (Myasuka) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java:356]: I think this method still useful even `requiresLegacySynchronousTimerSnapshots` could be removed, heap state backend should return false for this method. And we could remove the comment.

**Review by Myasuka — APPROVED** [review]
LGTM except for one line comment.

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java (hunk 1 above)
Comment (rkhachatryan) [reply flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java:356]: Which method do you mean?
(I'm not removing any methods in this PR)

Comment (Myasuka) [reply flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java:356]: I mean the `isStateImmutableInStateBackend` method. I think this comment description dose not match the role which `isStateImmutableInStateBackend` is playing. In other words, we cannot remove `isStateImmutableInStateBackend` even heap-based timers could work with RocksDB incremental snapshots.

Comment (rkhachatryan) [reply flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java:356]: Oh, I see. Yes, I agree and will remove the comment.
Thanks!

**Pull request merged**